Documentation/faq.md

```diff
@@ -8,10 +8,50 @@
 
 `advertise-urls` specifies the addresses etcd clients or other etcd members should use to contact the etcd server. The advertise addresses must be reachable from the remote machines. Do not advertise addresses like `localhost` or `0.0.0.0` for a production setup since these addresses are unreachable from remote machines.
 
+### Deployment
+
+#### Why odd number of cluster size?
+
+A quorum is the majority of nodes in a cluster. When n is the cluster size, quorum is (n/2)+1. 2 is the quorum both for 3 and 4. The primary goal of etcd is provide reliability and availability. etcd requires the majority (`n/2 + 1` when `n` is the cluster size) to be available to operate. Thus, adding another node to 3-node cluster does not add any value in terms of fault tolerance. And adding 1 node to single-node cluster increases the quorum size and causes leader election; if a wrong node were added by mistake, the cluster would lose its leader and not be able to make any progress. Always start with an odd number of nodes.
+
+#### What is maximum cluster size?
+
+Theoretically, there is no hard limit. However, we recommend no more than 7 nodes. [Google Chubby lock service][chubby], which is a similar project to etcd and widely deployed within Google for many years, suggests running 5 nodes. A 5-member etcd cluster can provide enough fault tolerance in most cases. While larger cluster provides better fault tolerance the write performance reduces since data needs to be replicated to more machines.
+
+#### What is failure tolerance?
+
+etcd cluster continues to operate as long as quorum of cluster can communicate with each other. etcd automatically handles transient failures (e.g. network partitions and delays) while ensuring the safety properties in Raft. etcd can also survive hardware failures (e.g. power outages) by persisting Write-Ahead-Log files onto disk, which contain all stream of events that happened. User just needs to restart the node with same configuration, and etcd will automatically replay its log to the exact point of time before failure.
```

Comment:
An etcd cluster operates so long as a member quorum can be established. If quorum is lost through transient network failures (e.g., partitions), etcd automatically and safely resumes once the network recovers and restores quorum; Raft enforces cluster consistency. For power loss, etcd persists the Raft log to disk; etcd replays the log to the point of failure and resumes cluster participation. For permanent hardware failure, the node may be removed from the cluster through [runtime reconfiguration](op-guide/runtime-configuration.md).